plugin/evm/vm.go

```diff
@@ -249,11 +249,12 @@ type VM struct {
 
 	config Config
 
-	chainID     *big.Int
-	networkID   uint64
-	genesisHash common.Hash
-	chainConfig *params.ChainConfig
-	ethConfig   ethconfig.Config
+	chainID          *big.Int
+	networkID        uint64
+	genesisHash      common.Hash
+	chainConfig      *params.ChainConfig
+	chainConfigExtra *params.ChainConfigExtra
```

Comment:
Thinking very defensively: there are occasions that geth code will reuse an object by resetting configuration. Is it safer to only carry the `ChainConfig` and always extract the `ChainConfigExtra` from it?

Reply:
I think even a function that returns Extra would be sufficient. IMO in plugin pkg (or generally in custom code) we should reduce wrapping the chainConfig and if possible directly call VM's functions to fetch it with wrapped form.